### DISCUSSION TITLE
How can I use this template for my C# telegram bot ?

### DISCUSSION THREAD

**Discussion by asindarov** [revenkroz/telegram-web-app-bot-example#10] [Q&A] [answered]
I want to create telegram bot which uses the telegram's latest web app technology. When I search for telegram web app examples on the internet I just come across this repository template but I do not know how I can integrate this template with my C# telegram bot. I need help on this. Can you please provide some resources or guide on that ?

**Comment by revenkroz** [comment, marked as answer]
Well, in general, the WebApp is just a website page. The bot should display it at the click of a button (which you can set up in the bot's settings), or send a keyboard with a button (there is an example in the README.md of this repository). 

The main question is how you want to organize the interaction of your bot written in C# with the user via WebApp. Perhaps you don't need to make any changes to the bot's code at all. You just need to host the page somewhere. 

The [README.md](https://github.com/revenkroz/telegram-web-app-bot-example/blob/master/README.md) seems to me to contain an excellent guide to the initial setup of the bot. Then you can build on it later.

**Reply by asindarov** [reply]
yeah, got it thanks for the response. However, what I really do not understand is what I should do to actually show the web app to the user in the telegram bot should I send some kind of a POST request to telegram api if yes, can you please send me the api link to do that ?

**Reply by asindarov** [reply]
I successfully initialized web app inside my telegram bot. Turns out I got your README.md wrong, now I make it work thanks!

**Reply by revenkroz** [reply]
Great to hear!
